### PR TITLE
dialogs: Create audit entries for dialog answers

### PR DESCRIPTION
**Pull request by vinzenz**
Previously dialog answers have not been tracked in the audit. This patch
introduces a new event type dialog-answer that will be emitted every
time a dialog answer is requested or received. This will report any
answers that are provided through answerfile or interactive reply.

Signed-off-by: Vinzenz Feenstra <vfeenstr@redhat.com>

### CODE REVIEW

**Comment by vinzenz** [comment]
e2e tests

**Comment by AloisMahdal** [comment]
For testing; how do i get the entry out of the db?

A totally wild guess

    SELECT data FROM audit
    WHERE event = 'dialog-answer-result';

?

**Comment by ZhukovGreen** [comment]
Please add a unit tests

**Comment by pirat89** [comment]
@vinzenz pls add unit tests. I will test it later this day manually but code looks pretty clear.

**Comment by vinzenz** [comment]
> For testing; how do i get the entry out of the db?
> 
> A totally wild guess
> 
> ```
> SELECT data FROM audit
> WHERE event = 'dialog-answer-result';
> ```
> 
> ?
```sql
SELECT data FROM audit
WHERE event = 'dialog-answer';
```

**Comment by AloisMahdal** [comment]
+1 for making it work with snactor, if possible, right away. we don't have test api to do it otherwise now.

**Comment by pirat89** [comment]
> +1 for making it work with snactor, if possible, right away. we don't have test api to do it otherwise now.

That's probably off-topic in this PR. As it could require more stuff to do in snactor. This PR is just about adding the answers into the leapp.db. Test is simple in such case - just run leapp and see whether you have correct input in the database.

File: tests/scripts/test_answerstore.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    a.translate(MockDialogBoolScope)
+    assert store['boolscope']['key1'] is True
+    assert store['boolscope']['key2'] is False
+    assert store['scope1']['key1'] == 'scope1.key1'
+    assert store['scope1']['key2'] == 'scope1.key2'
+    assert store['scope2']['key1'] == 'scope2.key1'
+    assert store['scope2']['key2'] == 'scope2.key2'
+
+
+def test_answerfile_generate(tmpdir, answerfile):
```
Comment (ZhukovGreen) [inline tests/scripts/test_answerstore.py:287]: some conditions of the tested function is not tested (i.e.  if hasattr(component, 'choices'):, if answer:)

**Comment by AloisMahdal** [comment]
@pirat89 "off-topic in this PR. As it could require more stuff to do in snactor" is a non-sequitur.

The test you are suggesting is what we'll fall back if there's no other option but that's not a reliable one as it relies on state of leapp-repository, and also it's not extensible.

**Comment by pirat89** [comment]
> @pirat89 "off-topic in this PR. As it could require more stuff to do in snactor" is a non-sequitur.
> 
> The test you are suggesting is what we'll fall back if there's no other option but that's not a reliable one as it relies on state of leapp-repository, and also it's not extensible.

Hmm.... Are you aware that you need to write tests using leapp anyway? Becuase for leapp and snactor there is kind of differend codebase.. Including fact that some parts related to answers is exclusive only for leapp?.... The crucial thing for us isn't fact whether answers are loaded and stored using snactor. That's nice-to-have for developers. But crucial is whether the stuff is there when leapp is executed. Because the main reason for these answers in the db is for debugging purposes.

So from that point, you doesn't have to rely on leapp-repository. You can deliver to leapp everything you have to on your own.

**Comment by AloisMahdal** [comment]
As I said, we don't have test api to do that, though. :/

We'll fall back to verification via leapp-repository now.

**Comment by kokesak** [comment]
Verified: https://liver3.brq.redhat.com/jenkins/job/oamg-run-beaker/7657//console
Build:  
 leapp-0.10.0-0.202008051159Z.1110644.log_answers.PR651.el7_6
           leapp-repository-0.10.0-100.202008071306Z.286f83f.master.el7_6

**Comment by bocekm** [comment]
Artem, please create a separate task to improve the tests per your suggestions. Thanks.

**Pull request merged**